.github/workflows/build-wheel.yml

```diff
@@ -128,7 +128,7 @@ jobs:
         if: github.event_name == 'workflow_dispatch'
         id: upload-wheel
         with:
-          name: ${{ steps.build-dist.outputs.wheel-name }}
+          name: ${{ steps.build-dist.outputs.wheel-name }}-${{ matrix.type }}
```

Comment:
@harupy @BenWilson2 are there any downstreams effects that this naming change could have? If so, @prithvikannan, it might be safer to only append to the name if the matrix type contains "databricks"

Reply:
I think this would only potentially break some dev workflows (probably very isolated), but I do think that we should only modify the discreet case that you're looking to isolate here. We should preserve a consistent PEP-425 compliant naming convention for the standard OSS version.

Reply:
Why did we need https://github.com/mlflow/mlflow/pull/13982? We can just use:

```
pip install 'mlflow[databricks] @ wheel path'
```

if we want to use databricks extra.

Reply:
You can confirm this by:

```
docker run python:3.9 pip install 'mlflow[databricks] @ https://files.pythonhosted.org/packages/f2/17/d24ec693ef27ce33f68414872f1f79ad3da62922edc6c6239a6de5336e5a/mlflow-2.18.0-py3-none-any.whl'
```

Reply:
Can we close this PR and revert https://github.com/mlflow/mlflow/pull/13982?